back/app/services/community_monitor_service.rb

```diff
@@ -12,9 +14,15 @@ def project_id
   end
 
   def project
-    return nil unless enabled?
+    return nil unless enabled? && project_id
+
+    @project ||= Project.find(project_id)
+  end
+
+  def phase
+    return nil unless enabled? && project_id
 
-    Project.find(project_id)
+    @phase ||= Phase.find_by(project_id: project_id)
   end
```

Comment:
Did a little refactoring too, to remove feature flags from other parts of the BE code